tldw_Server_API/tests/AuthNZ/test_audit_chain_integration.py

```
@@ -0,0 +1,248 @@
+                    rows = [dict(r) for r in await cur.fetchall()]
+
+            # Build verification events from stored data
+            verify_events = []
+            for row in rows:
+                verify_events.append({
+                    "action": row.get("action", ""),
+                    "user_id": row.get("context_user_id"),
+                    "timestamp": row.get("timestamp", ""),
+                    "detail": row.get("event_type", ""),
+                    "chain_hash": row.get("chain_hash", ""),
+                })
```

Ruff (0.15.6)

[warning] Use a list comprehension to create a transformed list

(PERF401)